gcc/rust/expand/rust-macro-expand.cc

```diff
@@ -109,8 +110,13 @@ MacroExpander::expand_decl_macro (Location invoc_locus,
       return AST::Fragment::create_error ();
     }
 
-  return transcribe_rule (*matched_rule, invoc_token_tree, matched_fragments,
-			  semicolon, peek_context ());
+  std::map<std::string, MatchedFragmentContainer *> matched_fragments_ptr;
+
+  for (auto &ent : matched_fragments)
+    matched_fragments_ptr.emplace (ent.first, ent.second.get ());
```

Comment:
Is there no way to avoid this and instead make changes to `transcribe_rule` and `SubstituteCtx` to accept an `std::map<std::string, std::unique_ptr<MatchedFragmentContainer>>`?

Reply:
Not without some finagling. Each repetition substitution would have to either clone ```MatchedFragmentContainer```s or temporarily move them out of shared state.